### PR TITLE
feat: build delegate project transaction

### PR DESCRIPTION
**Pull request by sk-yagi**

### CODE REVIEW

**Comment by sk-umiuma** [comment]
Could you split the delegations into multiple batches with given script size limit per tx? The tx is gonna fail given large enough inputs.

**Comment by sk-yagi** [comment]
> Could you split the delegations into multiple batches with given script size limit per tx? The tx is gonna fail given large enough inputs.

Sure, I will add that in upcoming PR as an optimization.

File: src/transactions/project/delegate.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  protocolParamsUtxo: UTxO;
+  allGenericSomething: GenericDelegateProjectParams[];
+  poolId: PoolId;
+};
+
+// TODO: Better naming!!!
+export type GenericDelegateProjectParams = {
+  projectUtxo: UTxO;
+  projectScriptVScriptUtxo: UTxO;
+  authorizedAddress: Address; // Either `StakingManager` or `GovernorAddress`
```
Comment (sk-saru) [inline src/transactions/project/delegate.ts:16]: This param is the same for all projects

File: src/transactions/project/delegate.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const projectStakingCredential =
+      lucid.utils.scriptHashToCredential(projectSvScriptHash);
+    const rewardAddress = lucid.utils.credentialToRewardAddress(
+      projectStakingCredential
+    );
+
+    tx = tx.addSigner(txParams.authorizedAddress);
+    tx = tx.readFrom([txParams.projectUtxo, txParams.projectScriptVScriptUtxo]);
+    tx = tx.registerStake(rewardAddress);
+    tx = tx.delegateTo(rewardAddress, poolId, Data.void());
```
Comment (sk-saru) [inline src/transactions/project/delegate.ts:44]: sequence dot

Comment (sk-saru) [inline src/transactions/project/delegate.ts:44]:
```suggestion
    tx = tx
      .addSigner(txParams.authorizedAddress)
      .readFrom([txParams.projectUtxo, txParams.projectScriptVScriptUtxo])
      .registerStake(rewardAddress)
      .delegateTo(rewardAddress, poolId, Data.void());
```

**Review by sk-saru — APPROVED** [review]

**Review by sk-saru — APPROVED** [review]

**Pull request merged**